### DISCUSSION TITLE
facing to download user profile exhaust and course progress exhaust and other reports.

### DISCUSSION THREAD

**Discussion by kamal-singhal** [Sunbird-Lern/Community#124] [Issues] [answered]
Hi sunbird team,

We got the issue raised by user that they are facing to download user profile exhaust and course progress exhaust and other reports.
They have requested about 2 weeks ago but haven't received the report yet.

We are trying to resolve these issues but while debugging the issues, following queries are not getting answered
1. once the report generate request sent by the user then Which table get the request Records inserted?
2. Some users can’t submit request generation again, it is showing processing, how can we change the status from backend so that we enable user again to submit a new request,
is there something that we can check in Database?  
4. Can you please tell us the which jobs are related for the below exhaust report and how to check the same.
  	1. Question set report
 	2. user profile exhaust report
	3. course process exhaust report

**Comment by kamal-singhal** [comment]
Hi team,

This is very critical issue. you are requested to please help to us.

**Reply by kumarks1122** [reply]
@kamal-singhal : Job request gets stored in postgres DB. postgres DB details can be found in the analytics service config.
table name: `{{env}}_job_request`
DB name: `analytics`
Also, user can submit the request for a batch only once within the same day.
Please find the job name respective to reports
```
Question set report: ResponseExhaustJobV2 (response-exhaust)
user profile exhaust report: UserInfoExhaustJob (userinfo-exhaust)
course process exhaust report: ProgressExhaustJob (progress-exhaust)
```
Kindly check the cronjobs or respective Jenkins job(for cluster mode) to check whether the job got scheduled to run everyday.

Also please check the logs for the spark execution.
cc: @Krishnaj20 @anandvarada @vinukumar-vs

**Reply by kamal-singhal** [reply]
Thank you for your response.

**Reply by kamal-singhal** [reply]
Hi team, 

@kumarks1122 : Thank you for your last response. as per you given solution, we have checked into postgres DB as well as checked the log of the jobs. We rectified the error which got in log and deleted the old request id in the postgres db. so that user will submit new fresh request for the same reports.

So now some users have able to  re-submit the request and able to download the report but still some users have sent a request for exhaust report but still showing status of report is "Processing". we didn't understand following things. please help us to solve these below points.
1.  User had sent a request to generate exhaust report ago 5 days but still report status is showing as "**Processing**". can you tell us why till now status is showing "Processing". 
2.  Can you tell us debugging steps for above use case.

for your reference, i have attached some screen shots. kindly look into this and need help us.
![VNU7RxTdkHwvGvjgqBhjUHtSqZAhXcHN7g](https://github.com/Sunbird-Lern/Community/assets/125978084/01e832d7-f16c-47a3-b7a8-9e78c1c8184d)
![Pdtd-qsZeZJLc7MzX5WbmPJEHQcQCBKxMQ](https://github.com/Sunbird-Lern/Community/assets/125978084/2ef6c168-ae93-475e-9fe4-155ac4085639)
![k-CXFkLmfCcUoOsYKivPefEim-6h2fLj4A](https://github.com/Sunbird-Lern/Community/assets/125978084/8c888e0d-ce46-44dc-94b2-cecea22fdd65)

**Reply by kumarks1122** [reply]
@kamal-singhal Those reports which are in processing state might have got failed due to the issue. Kindly update them to the `SUBMITTED` state and rerun the job.

**Reply by kamal-singhal** [reply]
Hi @kumarks1122 

Thanks for the Response. Can you tell us what problem caused this report to fail. Because this problem is not coming for only one user, this problem is coming for many users so we need permanent solution that why report is failing.

**Reply by kumarks1122** [reply]
@kamal-singhal : In production, exhaust reports run in parallel execution mode. So, the requests which were in each execution which got failed would have been still in `processing` state.

**Reply by AmiableAnil** [reply]
@kamal-singhal Please let us know if you were able to resolve your issue or let us know you need any thing else to close this issue.
cc:@rhwarrier @Krishnaj20

**Reply by kumarks1122** [reply]
@kamal-singhal : Can you please verify job execution log for whether the spark job is throwing the below exception?
```
com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0]
```
Can you also check joblog for that exception?
If the exception is there, can you take the request id from that log object.
Check for that request, `processed_batches` column value is `""` (empty string) or null?
cc: @reshmi-nair @AmiableAnil @vinukumar-vs

**Reply by Jayaprakash8887** [reply, marked as answer]
@kamal-singhal Please mark this thread as answered if there are no further queries.

**Comment by kamal-singhal** [comment]
Answered.
